lib/galaxy/model/migrations/util.py

```diff
@@ -10,24 +10,27 @@
 
 
 def drop_column(table_name, column_name):
+    if context.is_offline_mode():
+        return _handle_offline_mode(f"drop_column({table_name}, {column_name})", None)
+
     with op.batch_alter_table(table_name) as batch_op:
         batch_op.drop_column(column_name)
 
 
 def column_exists(table_name, column_name):
-    if is_offline_mode():
-        msg = (
-            "This script is being executed in offline mode and cannot connect to the database. "
-            f"Therefore, `column_exists({table_name}, {column_name})` returns `False` by default."
-        )
-        log.debug(msg)
-        return False
+    if context.is_offline_mode():
+        return _handle_offline_mode(f"column_exists({table_name}, {column_name})", False)
+
     bind = op.get_context().bind
     insp = inspect(bind)
     columns = insp.get_columns(table_name)
     return any(c["name"] == column_name for c in columns)
 
 
-def is_offline_mode():
-    cmd_opts = context.config.cmd_opts
-    return cmd_opts and cmd_opts.sql
+def _handle_offline_mode(code, return_value):
+    msg = (
+        "This script is being executed in offline mode and cannot connect to the database. "
+        f"Therefore, `{code}` returns `{return_value}` by default."
+    )
+    log.debug(msg)
```

Comment:
Maybe log at a higher level?

Reply:
Would `log.info` be a more appropriate choice? It's not an error: it happens when you do a dry run of an upgrade or a downgrade, and one of the revision scripts tries to access the database - so it's purely informational.

Reply:
`log.info` seems a good choice then.

Reply:
@nsoranzo if there's nothing else, would you mind approving if it lookss OK? I want to merge it before it conflicts :)